pkg/agent/memberlist/cluster_test.go

```diff
@@ -0,0 +1,596 @@
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes-16          	   52707	     22412 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes-16           	  538476	      2273 ns/op
+//PASS
+//BenchmarkCluster_ShouldSelectHitNodeByConsistentHash
+//BenchmarkCluster_ShouldSelect/select_node_from_1000_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_1000_alive_nodes-16         	12263878	        95.5 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes-16          	13036746	       103 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes-16           	14923483	        77.5 ns/op
+//PASS
+func BenchmarkCluster_ShouldSelect(b *testing.B) {
+	genNodes := func(n int) []string {
+		nodes := make([]string, n)
+		for i := 0; i < n; i++ {
+			nodes[i] = fmt.Sprintf("node-%d", i)
+		}
+		return nodes
+	}
+
+	benchmarkCases := []struct {
+		name       string
+		nodes      []string
+		egressName string
+		myNode     string
+	}{
+		{
+			name:       "select node from 10000 alive nodes",
+			nodes:      genNodes(10000),
+			egressName: "egress-10",
+			myNode:     "node-10",
+		},
+		{
+			name:       "select node from 1000 alive nodes",
+			nodes:      genNodes(1024),
+			egressName: "egress-10",
+			myNode:     "node-10",
+		},
+		{
+			name:       "select node from 100 alive nodes",
+			nodes:      genNodes(128),
+			egressName: "egress-10",
+			myNode:     "node-10",
+		},
+		{
+			name:       "select node from 10 alive nodes",
+			nodes:      genNodes(8),
+			egressName: "egress-10",
+			myNode:     "node-10",
+		},
+	}
+
+	for i := range benchmarkCases {
+		bc := benchmarkCases[i]
+		b.Run(fmt.Sprintf("%s-hitNodeByConsistentHash", bc.name), func(b *testing.B) {
+			b.ResetTimer()
+			conHash := consistenthash.New(1, nil)
+			conHash.Add(bc.nodes...)
+			for i := 0; i < b.N; i++ {
+				hitNodeByConsistentHash(conHash, bc.egressName, bc.myNode)
+			}
+		})
+		b.Run(fmt.Sprintf("%s-hitNodeBySortHash", bc.name), func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				hitNode(bc.nodes, bc.egressName, bc.myNode)
+			}
+		})
+	}
+}
+
+func genRandomStr(num int) string {
+	buf := make([]byte, num)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return ""
+	}
+	return fmt.Sprintf("%016x", buf)
+}
+
+func TestCluster_ShouldSelect(t *testing.T) {
+	nodes := []string{"node1", "node2", "node3"}
+
+	genLocalNodeCluster := func(nodeNme string) *Cluster {
+		cluster := &Cluster{
+			memberList: &memberlist.Memberlist{},
+			NodeConfig: &config.NodeConfig{
+				Name: nodeNme,
+			},
+			conHash: consistenthash.New(1, nil),
+		}
+		return cluster
+	}
+
+	node1Cluster := genLocalNodeCluster("node1")
+
+	monkey.PatchInstanceMethod(
```

Comment:
Why it needs to patch the method? I didn't see the method will affect result of `ShouldSelect`.
To fake a method's return value, function variable, interface with a fake implementation, gomock are more common and already adopted by Antrea and should satisfy all scenarios. Maybe try use them to avoid introducing another mechanism.

Reply:
done, forget to delete the test case, as previous version code need a patch here, I will have a look at gomock as you recommend